### PR TITLE
reposync: import GPG keys to RPM DB individually (bsc#1217003)

### PR DESCRIPTION
**Pull request by meaksh**
## What does this PR change?

This PR refactors the code around GPG keys importing from Spacewalk keyring to the RPM DB that is used by Zypper during `spacewalk-repo-sync` execution:

- GPG keys are processed individually, instead of all in the same file.
- Continue processing keys even if there are errors in some of them. 
- Enhance related import logging.

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [x] **DONE**

## Test coverage
- No tests: **already covered at accepteance tests. No unit tests for GPG keys import**

- [x] **DONE**

## Links

Issue(s):
- https://github.com/SUSE/spacewalk/issues/23150
- https://github.com/uyuni-project/uyuni/issues/8176 (?)

Ports(s): https://github.com/SUSE/spacewalk/pull/23629

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

# Before you merge

Check [How to branch and merge properly](https://github.com/uyuni-project/uyuni/wiki/How-to-branch-and-merge-properly)!

### CODE REVIEW

**Review by vzhestkov — DISMISSED** [review]
Looks good for me. Thanks for handling it :+1:

**Review by agraul — APPROVED** [review]
LGTM

File: python/spacewalk/satellite_tools/repo_plugins/yum_src.py [hunk 1]
```diff
@@ -147,22 +148,47 @@ def __synchronize_gpg_keys(self):
...
+                "--export",
+                "--export-options",
+                "export-clean",
+                "--with-colons",
+                "-a",
+                "--output",
+                all_keys_file,
+            ]
+            _log_command(args)
+            # pylint: disable-next=subprocess-run-check
```
[the remark this answers is not in the document]
Comment (meaksh) [reply python/spacewalk/satellite_tools/repo_plugins/yum_src.py]: fixed

File: python/spacewalk/satellite_tools/repo_plugins/yum_src.py [hunk 2]
```diff
@@ -198,63 +229,82 @@ def __synchronize_gpg_keys(self):
...
                         # pylint: disable-next=consider-using-f-string
-                        "rpm --dbpath {} -e gpg-pubkey-{}-{}".format(
-                            REPOSYNC_ZYPPER_RPMDB_PATH, key, zypper_gpg_keys[key]
-                        )
-                    )
+                        "gpg-pubkey-{}-{}".format(key, zypper_gpg_keys[key]),
+                    ]
+                    _log_command(args)
+                    # pylint: disable-next=subprocess-run-check
+                    subprocess.run(args)
```
[the remark this answers is not in the document]
Comment (meaksh) [reply python/spacewalk/satellite_tools/repo_plugins/yum_src.py]: fixed

**Review by agraul — APPROVED** [review]

**Pull request merged**